♻️ Refactor gist list to use graphQL

gleich opened the pull request:
This PR resolves #1712.

gleich committed:
♻️ Refactor gist list to use graphQL

Signed-off-by: Matthew Gleich <matthewgleich@gmail.com>

mislav reviewed (changes requested):
This looks fantastic; thank you!

I think that the stubs in tests need to be updated to compensate for the switch to GraphQL

gleich commented:
Yep. I was looking into it yesterday and I'll try to get it working. Haven't really worked with the testing side of things with graphql before so I might need a little help.

gleich committed:
🐛 Fix stubs

Signed-off-by: Matthew Gleich <email@mattglei.ch>

gleich committed:
♻️ Refactor to only pass time in once for test

Signed-off-by: Matthew Gleich <email@mattglei.ch>

mislav committed:
Fix parsing `gist list --public/--secret` flags

It's not sufficient to use `Changed("public")` to test if a boolean flag
was activated, since the user might have passed `--public=false`.
Instead, check the true value of the flag.

The `--public` and `--secret` flags should be mutually exclusive, so now
if both are activated, `--secret` takes precedence.

mislav committed:
Port listing gists to githubv4 GraphQL client

- Fetching more than 100 gists is now supported
- The GraphQL query name is now `GistList` instead of `ListGists` for
  consistency with other queries
- Avoid fetching unnecessary Files fields
- Gists are now rendered in the order that the API returned them in
- The gist timestamp for machine-readable output is now rendered in
  RFC3339 format instead of in `time.Time.String()` format which is only
  meant for debugging and is not considered stable
- Ensure newlines in gist description are rendered as spaces

mislav reviewed (approved):
@Matt-Gleich Thanks for the updates!

@vilmibm I've added a number of changes to this branch:

- Ported the GraphQL query to "githubv4" adapter
- Fetching more than 100 gists is now supported
- The GraphQL query name is now `GistList` instead of `ListGists` for consistency with other queries
- Avoid fetching unnecessary Files fields
- Gists are now rendered in the order that the API returned them in (CreatedAt desc) instead of reordered by UpdatedAt timestamp - should we consider displaying the CreatedAt timestamp instead of UpdatedAt?
- The gist timestamp for machine-readable output is now rendered in RFC3339 format instead of in `time.Time.String()` format which is only meant for debugging and is not considered stable
- Ensure newlines in gist description are rendered as spaces
- It's not sufficient to use `Changed("public")` to test if a boolean flag was activated, since the user might have passed `--public=false`. Instead, check the true value of the flag.
- The `--public` and `--secret` flags should be mutually exclusive, so now if both are activated, `--secret` takes precedence.

khader8 commented:
api/queries_issue.go

Matt-Gleich commented:
?
